### PR TITLE
update importimage, projects, rsh, req_project

### PR DESCRIPTION
**Pull request by juanvallejo**
Updates `importimage.go`, `projects.go`, `rsh.go`, `project.go`, and `request_project.go`

cc @deads2k

### CODE REVIEW

**Comment by juanvallejo** [comment]
cc @deads2k

File: pkg/oc/cli/cmd/importimage.go [hunk 1]
```diff
@@ -41,39 +40,17 @@ var (
...
-	cmd.Flags().BoolVar(&opts.DryRun, "dry-run", false, "Fetch information about images without creating or updating an image stream.")
-	cmd.Flags().BoolVar(&opts.Scheduled, "scheduled", false, "Set each imported Docker image to be periodically imported from a remote repository. Defaults to false.")
-	opts.Insecure = cmd.Flags().Bool("insecure", false, "If true, allow importing from registries that have invalid HTTPS certificates or are hosted via HTTP. This flag will take precedence over the insecure annotation.")
-
-	return cmd
-}
-
 // ImageImportOptions contains all the necessary information to perform an import.
 type ImportImageOptions struct {
+	genericclioptions.IOStreams
```
Comment (deads2k) [inline pkg/oc/cli/cmd/importimage.go]: bottom of struct

File: pkg/oc/cli/cmd/importimage.go [hunk 2]
```diff
@@ -87,24 +64,52 @@ type ImportImageOptions struct {
...
-	out         io.Writer
-	errout      io.Writer
 	imageClient imageclient.ImageInterface
 	isClient    imageclient.ImageStreamInterface
 }
 
+func NewImportImageOptions(name string, streams genericclioptions.IOStreams) *ImportImageOptions {
+	return &ImportImageOptions{
+		IOStreams:       streams,
+		CommandName:     name,
```
Comment (deads2k) [inline pkg/oc/cli/cmd/importimage.go]: do I even want to know why they need this?

Comment (deads2k) [inline pkg/oc/cli/cmd/importimage.go]: surely this can be set by the cobra cmd passed in apply.

File: pkg/oc/cli/cmd/projects.go [hunk 3]
```diff
@@ -30,13 +29,21 @@ type ProjectsOptions struct {
...
 
 	DisplayShort bool
+
+	genericclioptions.IOStreams
+}
+
+func NewProjectsOptions(name string, streams genericclioptions.IOStreams) *ProjectsOptions {
+	return &ProjectsOptions{
+		IOStreams:   streams,
+		CommandName: name,
```
Comment (deads2k) [inline pkg/oc/cli/cmd/projects.go]: inside of cobra command?

File: pkg/oc/cli/cmd/projects.go [hunk 4]
```diff
@@ -113,19 +112,20 @@ func (o *ProjectsOptions) Complete(f kcmdutil.Factory, args []string, commandNam
...
 
 // RunProjects lists all projects a user belongs to
 func (o ProjectsOptions) RunProjects() error {
-	config := o.Config
-	clientCfg := o.ClientConfig
-	out := o.Out
-
 	var currentProject string
-	currentContext := config.Contexts[config.CurrentContext]
+	currentContext := o.Config.Contexts[o.Config.CurrentContext]
```
Comment (deads2k) [inline pkg/oc/cli/cmd/projects.go]: You changed this from a shallow copy to the same object.  Are you sure that's safe/correct?

File: pkg/oc/cli/cmd/request_project.go [hunk 5]
```diff
@@ -149,6 +155,7 @@ func (o *NewProjectOptions) Run() error {
 		o.ProjectOptions.ProjectName = project.Name
 		o.ProjectOptions.ProjectOnly = true
 		o.ProjectOptions.SkipAccessValidation = true
+		o.ProjectOptions.IOStreams = o.IOStreams
```
Comment (deads2k) [inline pkg/oc/cli/cmd/request_project.go]: This block seems weird, but I guess we don't have to fix it here.

File: pkg/oc/clusterup/docker/openshift/project.go [hunk 6]
```diff
@@ -34,9 +34,9 @@ func CreateProject(f genericclioptions.RESTClientGetter, name, display, desc, ba
 		Client: projectClient.Project(),
 
 		ProjectOptions: &cmd.ProjectOptions{PathOptions: pathOptions},
-		Out:            ioutil.Discard,
+		IOStreams:      genericclioptions.IOStreams{Out: ioutil.Discard, ErrOut: ioutil.Discard},
```
Comment (deads2k) [inline pkg/oc/clusterup/docker/openshift/project.go]: genericclioptions.NewDiscardingIOStreams

**Comment by deads2k** [comment]
nits, lgtm otherwise.  Self tag when nits are fixed

/approve

File: pkg/oc/cli/cmd/importimage.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/importimage.go]: Ah, was not being used - removed

File: pkg/oc/cli/cmd/projects.go (hunk 4 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/projects.go]: Thanks, will revert this change

**Comment by juanvallejo** [comment]
Thanks, nits addressed. Tagging

**Pull request merged**